core/src/main/scala/akka/kafka/CommitterSettings.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka
+import java.util.concurrent.TimeUnit
+
+import akka.actor.ActorSystem
+import com.typesafe.config.Config
+
+import scala.concurrent.duration._
+
+object CommitterSettings {
+
+  def apply(actorSystem: ActorSystem): CommitterSettings =
+    apply(actorSystem.settings.config.getConfig("akka.kafka.committer"))
+
+  def apply(config: Config): CommitterSettings = {
+    val maxBatch = config.getLong("max-batch")
+    val interval = config.getDuration("interval", TimeUnit.MILLISECONDS).millis
+    new CommitterSettings(maxBatch, interval)
+  }
+}
+
+class CommitterSettings(
```

Comment:
Mark the constructor as `private`, please.
```suggestion
class CommitterSettings private (
```